### PR TITLE
Dimensions to be altered by UDF exported in JSON

### PR DESCRIPTION
**Pull request by pramitghosh**
Referencing Open-EO/openeo-udf#7, I have modified the structure of the JSON (which was suggested by @huhabla) being sent to the UDF service with the POST request a bit so that it contains the dimension which is being modified by the UDF. This could then be parsed by the R UDF service.

@flahn could you please check if the JSON is being generated correctly for `aggregate_time`. Hopefully, it doesn't break anything else ;-)

### CODE REVIEW

**Comment by flahn** [comment]
Please notice that I have reworked the Dockerfile and installation scripts. I won't be able to include your changes.

**Comment by pramitghosh** [comment]
Yes, right. But, hopefully `Udf-class.R` could be merged. Maybe I should create a new branch without the changes in the Dockerfile and create a new pull request (?)

File: R/processes.R [hunk 1]
```diff
@@ -322,7 +322,7 @@ aggregate_time = Process$new(
     # prepare paths
     udf_transaction = prepare_udf_transaction(user,script,job$job_id)
     
-    udf_transaction$prepareExportData(collection,export_type=c("json","file"))
+    udf_transaction$prepareExportData(collection,export_type=c("json","file"), dim_mod = names(which(self$modifier == TRUE)))
```
Comment (flahn) [inline R/processes.R]: during debugging i found that in the context of execution of the graph `sel$modifier` is not available. Also the dimension modifier consists of two dimension lists (for adding and removing dimensions)

**Comment by flahn** [comment]
I tried to merge your changes from https://github.com/Open-EO/openeo-r-backend/pull/15/commits/ae5b7daa94c515563afb18a4138e5eaca546bf2f. Commit https://github.com/Open-EO/openeo-r-backend/commit/41e5f06950c75f41b03ea8591b6e77ed52f02db4 should do it. However there have been some issues with your code. I hope that it is correct that you wanted to pass on the dimensions that needs to be removed.

**Comment by pramitghosh** [comment]
Yes, you're right that the idea is to pass on the dimension to be modified (to be removed in case the UDF is a reducer) to the UDF service along with the JSON. The UDF service would then [set the dimension to be modified](https://github.com/pramitghosh/openEO.R.UDF/blob/83fd3c6f4b7d126ceaf11a7667c6dccaf1981196/R/RESTful_services.R#L342) based on the value of `code/alt_dim` in the incoming JSON. I will test this against endpoint `/udf` and get back to you soon. Thanks!

**Comment by pramitghosh** [comment]
When I run the Docker container of this repository (develop branch), I receive the following warnings and error:
```
Warning messages:
1: replacing previous import ‘magrittr::extract’ by ‘raster::extract’ when loading ‘openEO.R.Backend’ 
2: replacing previous import ‘dplyr::select’ by ‘raster::select’ when loading ‘openEO.R.Backend’ 
3: replacing previous import ‘dplyr::intersect’ by ‘raster::intersect’ when loading ‘openEO.R.Backend’ 
4: replacing previous import ‘dplyr::union’ by ‘raster::union’ when loading ‘openEO.R.Backend’ 
> 
> config = ServerConfig()
> config$workspaces.path = "/var/openeo/workspace"
> config$initEnvironmentDefault()
Error: attempt to apply non-function
Execution halted
```

**Pull request closed without merging**